metrics/coverage_converter/docs/01_coverage_converter_design.md

```diff
@@ -0,0 +1,161 @@
+
+# Design
+
+> Explain and diagram the technical design
+
+## Main interfaces and classes
+
+The main purpose of this tool is to convert the given coverage report to the `CI integrations` readable one. To do so, we should create the following interfaces and abstract classes: 
+
+- The `CoverageConverterCommand` interface - the common code coverage command interface. 
+- The `CoverageConverter` interface that will represent the common interface for converting the specific coverage report format to the `CI integrations` coverage report format.
+
+Let's consider the class diagram explaining the main interfaces and dependencies between them: 
+
+![Interfaces diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_interfaces_diagram.puml)
+
+As we can see above, the `CoverageConverterCommand` uses the `CoverageConverter` to convert the specific coverage report to the `CoverageModel` class that represents the `CI integrations` coverage report.
+
+Let's consider the activity diagram of the coverage conversion process: 
+
+![Coverage conversion diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_conversion_activity.puml)
+
+To run the command line application, we should have a `CommandRunner` class. Let's name it `CoverageConverterRunner`. This class will extend `CommandRunner`. It will be the place where we'll add our specific `CoverageConverterCommand`s.
+
+Also, we should be able to pass the following arguments to the Coverage Converter tool: 
+
+- input - the file from where we'll read the specific coverage report;
+- output - the file to which we'll write the formatted coverage report;
+
+To be able to pass arguments to our application we should create a `CoverageArgumentsParser` that will take a `List<String>` as arguments and return the `CoverageConverterArguments` instance that will contain all given arguments.
+
+## Package structure
+
+Once we've finished with all main classes and interfaces, let's consider the package structure of the `Coverage Converter`: 
+
+> lib
+>   - arguments
+>     - model
+>     - parser
+>   - common 
+>     - command
+>     - runner
+>     - converter
+>   - specific_format
+>     - command
+>     - model
+>     - converter
+
+## Structure of the Coverage coverage report.
+
+Since we want to unify the coverage reports for the CI integrations, we should create a specific for CI integrations coverage report format. Currently, this report will contain only `total coverage percent`. Let's consider the JSON summary structure: 
+
+```
+{
+  pct: 0.6
+}
+```
+
+So, the JSON will contain only `pct` value that will represent the total coverage percent. Let's consider the class diagram of the coverage structure: 
```

Comment:
Should we add more details like data type, range, precision, etc..? Is it compatible with Istanbul format? Can it have other fields?